src/control/server/engine/config.go

```diff
@@ -741,20 +758,30 @@ func (c *Config) WithStorageIndex(i uint32) *Config {
 // WithProperEnvVarForPMDK sets PMDK related environment variables
 // according to actual DCPMem configuration.
 func (c *Config) WithProperEnvVarForPMDK() *Config {
-	if len(c.Storage.Tiers) > 0 {
-		if c.Storage.Tiers[0].Class == storage.ClassDcpm {
-			envVars, err := common.DeleteKeyValue(c.EnvVars, "PMEMOBJ_CONF")
-			if err == nil {
-				c.EnvVars = envVars
-			}
-			return c.WithEnvVarAbtThreadStackSize(minABTThreadStackSizeDCPM)
-		} else {
-			envVars, err := common.DeleteKeyValue(c.EnvVars, "ABT_THREAD_STACKSIZE")
-			if err == nil {
-				c.EnvVars = envVars
-			}
-			return c.WithEnvVarPMemObjSdsAtCreate(0)
-		}
+	if len(c.Storage.Tiers) == 0 {
+		return c
+	}
+
+	if c.Storage.Tiers[0].Class == storage.ClassDcpm {
+		return c.WithoutEnvVarForPMDK().WithEnvVarAbtThreadStackSize(minABTThreadStackSizeDCPM)
+	} else {
```

Comment:
you don't need the else clause immediately after a return

Reply:
Done